app/components/public/ticket-list.js

```diff
@@ -176,14 +200,32 @@ export default Component.extend(FormMixin, {
       this.send('togglePromotionalCode', this.code);
     }
   },
+  donationTicketsValidation: computed('donationTickets.@each.id', 'donationTickets.@each.minPrice', 'donationTickets.@each.maxPrice', function() {
+    let validationRules = {};
+    this.donationTickets.forEach(donationTicket => {
+      validationRules = merge(validationRules, {
+        [donationTicket.id]: {
+          identifier : donationTicket.id,
+          optional   : true,
+          rules      : [
+            {
+              type   : `integer[${donationTicket.minPrice}..${donationTicket.maxPrice}]`,
+              prompt : this.l10n.t(`Please enter a donation amount between ${donationTicket.minPrice} and ${donationTicket.maxPrice}`)
+            }
+          ]
+        }
+      });
+    });
+    return validationRules;
+  }),
```

Comment:
@niranjan94 I've generated one rule per ticket in this way, and merged it with the fields of validationRules in the end. It works fine now.